pkg/sdk/dto-builder-generator/main.go

```diff
@@ -204,8 +204,20 @@ func (gen *Generator) generateBuilderMethods(d *structDef) {
 	}
 
 	for _, field := range optionalFields {
-		gen.printf("func (s *%s) With%s(%s %s) *%s {\n", d.name, toTitle(field.name), field.name, field.typeString, d.name)
-		gen.printf("s.%s = %s\n", field.name, field.name)
+		gen.printf("func (s *%s) With%s(%s %s) *%s {\n", d.name, toTitle(field.name), field.name, strings.TrimLeft(field.typeString, "*"), d.name)
+
+		switch {
+		case field.typeString == "*string":
```

Comment:
I am not sure about this logic. It is a bit misleading for the caller. Also, usually we invoke it after doing the tf check if the value is ok - it's not ok for empty string, so I think it would used used hardly ever.

Reply:
Removed this case, re-generated builder methods for the external table, and adjusted the resource so It will only set those fields when they're set.